hal/src/nRF52840/usart_hal.cpp

```diff
@@ -297,11 +305,8 @@ class Usart {
         const ssize_t maxRead = CHECK(data());
         const size_t readSize = std::min((size_t)maxRead, size);
         CHECK_TRUE(readSize > 0, SYSTEM_ERROR_NO_MEMORY);
-        ssize_t r;
-        {
-            RxLock lk(uarte_);
-            r = CHECK(rxBuffer_.get(buffer, readSize));
-        }
+        RxLock lk(uarte_);
```

Comment:
Is it safe to move RxLock outside an inner block here? ( It appears it would not be released until after startReceiver finishes in some cases. )

Reply:
It should be fine. It will only postpone firing of RX_END interrupt, which wouldn't happen anyway immediately as that's the end of the whole RX DMA transfer (which is multiple bytes, not one).